components/common/site_name_and_description.jsx

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+export default class SiteNameAndDescription extends React.PureComponent {
+    static propTypes = {
+
+        /**
+         * Custom description of the site
+         */
+        customDescriptionText: PropTypes.string,
+
+        /**
+         * Set to true if instance is licensed; false as default
+         */
+        isLicensed: PropTypes.bool,
+
+        /**
+         * Site name; "Mattermost" as default
+         */
+        siteName: PropTypes.string,
+    };
+
+    static defaultProps = {
+        isLicensed: false,
+        siteName: 'Mattermost',
+    };
+
+    render() {
+        const {
+            customDescriptionText,
+            isLicensed,
+            siteName,
+        } = this.props;
+        let description = null;
+        if (isLicensed && customDescriptionText) {
+            description = customDescriptionText;
```

Comment:
Show custom site's description if the instance is licensed and `customDescriptionText` is not empty.